### PR TITLE
xio: migrate atomic_t to std::atomic<>

### PR DESCRIPTION
**Pull request by chardan**
Signed-off-by: Jesse Williamson <jwilliamson@suse.de>

### CODE REVIEW

File: src/msg/xio/XioMessenger.cc [hunk 1]
```diff
@@ -30,9 +30,9 @@
 #define dout_prefix *_dout << "xio."
 
 Mutex mtx("XioMessenger Package Lock");
-atomic_t initialized;
+std::atomic<unsigned> initialized = { 0 };
```
Comment (tchaikov) [inline src/msg/xio/XioMessenger.cc]: this should be `std::atomic<bool>`

**Comment by Adirl** [comment]
@tchaikov 
:+1:

**Comment by tchaikov** [comment]
@chardan could you squash these two commit into a single one?

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-07_16:25:35-rados-wip-sage-testing2-distro-basic-smithi/

**Pull request merged**